app/src/main/java/com/google/android/sambadocumentsprovider/provider/SambaDocumentsProvider.java

```diff
@@ -359,11 +429,46 @@ private Cursor buildErrorCursor(String[] projection, @StringRes int resId) {
         case Document.COLUMN_LAST_MODIFIED:
           row[i] = metadata.getLastModified();
           break;
+        case Document.COLUMN_ICON:
+          row[i] = metadata.getIconResourceId();
```

Comment:
Actually if we want to use the default icon, we should pass null. See [link](https://developer.android.com/reference/android/provider/DocumentsContract.Document.html#COLUMN_ICON).

MatrixCursor does the auto-boxing, so just make DocumentMetadata#getIconResourceId() return Integer. It won't cause any further side effect.